components/timeline/src/Timeline.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, ReactNode, createElement as h } from 'react';
+import { StandardProps, classBuilder } from '@not-govuk/component-helpers';
+
+import '../assets/Timeline.scss';
+
+type TimelineItem = {
+  action: string
+  by?: string
+  date: string
+  description: ReactNode
+}
+
+export type TimelineProps = StandardProps & {
+  /** Items is a compulsory property of the Timeline component.
+  Items consists of an array of item objects, each consisting of an
+  action, the name of the person who performed the action (optional),
+  the date (and time) the action was taken and a description of the action.
+  */
+  items: TimelineItem[]
+};
+
+export const Timeline: FC<TimelineProps> = ({
+  children,
+  classBlock,
+  classModifiers,
+  className,
+  items,
+  ...attrs
+}) => {
+  const classes = classBuilder(
+    'hods-timeline',
+    classBlock,
+    classModifiers,
+    className
+  );
+
+  return (
+      <div className={classes()}>
+        {items.map(e => (
+          <div className={classes('item')}>
+            <h2 className={classes('title')}>{e.action}</h2>
+            {!e.by ? null : (<p className={classes('by')}> by {e.by}</p>)}
+            <p className={classes('date')}>{e.date}</p>
```

Comment:
I wonder if this should be an HTML `<time>` element?: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time

@RobinKnipe: Any thoughts?

Reply:
The component I copied used the <time> element originally, but I removed it. I felt that it was more restrictive, forcing users to enter the datetime attribute in the ISO format required, and maybe more prone to errors as a result?

Reply:
Build-time errors are good.
Run-time errors are bad.

By being restrictive over your interface, you will help to turn run-time errors into build-time errors.

Of course, in this case, you are just passing the data straight through so the user has complete control. However we should be writing [Semantic HTML](https://en.wikipedia.org/wiki/Semantic_HTML) wherever  possible.

If the original component used it (do you have a link BTW?) I'd keep it in.

Reply:
Thanks @daniel-ac-martin - yes lots of thoughts, but communicated in the wrong place! Where I was questioning the use of `string` for the date type earlier, I've basically put some of the same kinds of things you've mentioned here.

I agree, the `<time>` element definitely should be used here. Its `datetime` attribute should be set with the most accurate ISO string possible (date/date+time/full timestamp/), and hopefully a timezone. The displayed text (child) should be more readable, and follow the style guide expectations around gov [dates](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#dates)/[times](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#times). I definitely see this effort being captured in a proper component though.

Reply:
I thought `<time>` was meant to be a magical HTML5 thing that automatically did the formatting for you.

Reply:
No, sadly. From what I've read it merely seems to split the declaration of an _exact_ timestamp, from the text that gets rendered. From my examples above, a [jsfiddle](https://jsfiddle.net/qwsjo13b/)

Alas, no _obvious_ magic. However the result _is_ a `time` element, so at least the exact information is there, and any browser add-on or a11y behaviour should be supported.

Reply:
FYI: New issue for the date/time component #69